protos/sigstore_trustroot.proto

```diff
@@ -84,3 +84,23 @@ message TrustedRoot {
         // A set of trusted timestamping authorities.
         repeated CertificateAuthority timestamp_authorities = 4;
 }
+
+// Environment acts a selector to filter down a trust root to a smaller one.
+// A policy should reference an environment, or embed one that can be used
+// during artifact verification to filter the global trust root into a one
+// only containing the relevant instances. An environment does not take
+// temporality into account when selecting the instances from a trust root.
+// Prior to performing verification the temporal aspect should be further
+// examined to reduce the set of instances.
+message Environment {
+        // The name of the environment.
+        string name = 1;
```

Comment:
I never know whether to stick the name inside the struct, or just have a map<string, Environment>

We should consider both options based on how we're planning to use this

Reply:
Agreed, but I usually tend to replicate the name into the struct, even if it's duplicated as then the struct on its own has _some_ kind of identifier.